internal/store/subscriber.go

```diff
@@ -37,13 +48,33 @@ func (l *subscriberList) Remove(s Subscriber) error {
 	for i, current := range l.subscribers {
 		if s == current.subscriber {
 			l.subscribers = append(l.subscribers[:i], l.subscribers[i+1:]...)
+			if l.setup {
+				current.maybeTeardown(context.TODO())
```

Comment:
Should we fill out this TODO or are you leaving it for now?

Reply:
good point! updated it to use a real context